### PR TITLE
chore: include token masters in recipients community requests

### PR DESCRIPTION
**Pull request by 0x-r4bbit**
Token masters will contribute in accepting/rejecting member requests the same way admins do, so users will have to send requests to them as well.

### CODE REVIEW

File: protocol/communities/community.go [hunk 1]
```diff
@@ -1189,15 +1201,29 @@ func canDeleteMessageForEveryonePermissions() map[protobuf.CommunityMember_Roles
 	return roles
 }
 
+func (o *Community) GetAdminsAndTokenMasters() []*ecdsa.PublicKey {
+	return append(o.GetMemberTokenMasters(), o.GetMemberAdmins()...)
+}
+
 func (o *Community) GetMemberAdmins() []*ecdsa.PublicKey {
```
Comment (mprakhov) [inline protocol/communities/community.go]: I think it is better to name it
GetAdminMembers
The same for GetTokenMasterMembers

File: protocol/messenger_communities.go [hunk 2]
```diff
@@ -907,9 +907,8 @@ func (m *Messenger) RequestToJoinCommunity(request *requests.RequestToJoinCommun
 		return nil, err
 	}
 
-	// send request to join also to community admins
-	communityAdmins := community.GetMemberAdmins()
-	for _, communityAdmin := range communityAdmins {
+	adminsAndTokenMasters := community.GetAdminsAndTokenMasters()
```
Comment (mprakhov) [inline protocol/messenger_communities.go]: what about owners without commuunity private key?

**Review by mprakhov — APPROVED** [review]
LGTM, few comments

File: protocol/messenger_communities.go (hunk 2 above)
Comment (0x-r4bbit) [reply protocol/messenger_communities.go]: Good point, will add!
Will they be able to handle requests to join atm?

Comment (mprakhov) [reply protocol/messenger_communities.go]: Not fully understood the question, but the owner without a community private key will behave the same as the admin

File: protocol/messenger_communities.go [hunk 3]
```diff
@@ -924,9 +924,8 @@ func (m *Messenger) RequestToJoinCommunity(request *requests.RequestToJoinCommun
 		return nil, err
 	}
 
-	// send request to join also to community admins
-	communityAdmins := community.GetMemberAdmins()
-	for _, communityAdmin := range communityAdmins {
+	adminsAndTokenMasters := community.GetPrivilegedMembers()
```
Comment (0x-r4bbit) [inline protocol/messenger_communities.go]: @mprakhov this includes owners as well now

File: protocol/communities/community.go [hunk 4]
```diff
@@ -1209,15 +1221,33 @@ func canDeleteMessageForEveryonePermissions() map[protobuf.CommunityMember_Roles
...
+func (o *Community) GetAdminMembers() []*ecdsa.PublicKey {
+	return o.getMembersByPredicate(o.IsMemberAdmin)
+}
+
+func (o *Community) GetTokenMasterMembers() []*ecdsa.PublicKey {
+	return o.getMembersByPredicate(o.IsMemberTokenMaster)
+}
+
+func (o *Community) GetOwnerMembers() []*ecdsa.PublicKey {
+	return o.getMembersByPredicate(o.IsMemberOwner)
```
Comment (mprakhov) [inline protocol/communities/community.go:1237]: do we need to filter out the owner, who is the control node in order not to send 2 requests to join?

**Comment by 0x-r4bbit** [comment]
There's a test that has been failing a bunch of times on CI but I couldn't reproduce locally yet. Rerunning CI tests again

**Comment by 0x-r4bbit** [comment]
This is now failing with another test.
So seems like this is flaky.

Need to investigate this more.
Only idea I have is that this could cause a problem because we're now sending the request to join to 
-> control node
-> then, not only admins, but also owners (which we don't know if they are control node as well)

File: protocol/communities/community.go [hunk 5]
```diff
@@ -1234,6 +1236,31 @@ func canDeleteMessageForEveryonePermissions() map[protobuf.CommunityMember_Roles
...
+func (o *Community) getMembersByPredicate(filterFn predicate) []*ecdsa.PublicKey {
+	result := make([]*ecdsa.PublicKey, 0)
+	members := o.GetMemberPubkeys()
+	for _, member := range members {
+		if filterFn(member) {
+			result = append(result, member)
+		}
+	}
+	return result
+}
```
Comment (mprakhov) [inline protocol/communities/community.go]: this is not needed anymore
I added `GetPrivilegedMembers` which will do the same

Comment (0x-r4bbit) [reply protocol/communities/community.go]: Removed!

**Pull request merged**